File: bup240594arreglos_y_matrices.java
import java.util.Arrays;

class bup240594arreglos_y_matrices{
    public static void main(String[] args){
        String[] nombres = {"Ana","Luis","Carlos","Marta"};
        String[] materias= {"Matematicas", "Fisica" , "Quimica" , "Biologia"};
        double[][] calificaiones = {
            {10,9,8,7},
            {9,7,6,8},
            {8,9,5,9},
            {7,8,9,10}
        };


        for (int i = 0; i < calificaiones.length; i++){
            System.out.print(Arrays.toString(calificaiones[i]));
            if(i < calificaiones.length){
                System.out.println(", \n");
            }
        } 
    }
}

File: bup240594arreglos_y_matrices.py
nombres = ['Ana', 'Luis', 'Carlos', 'Marta', 'Promedio']
materias = ['','Matemáticas', 'Física', 'Química', 'Biología']
calificaciones = [[10, 9, 8, 7],
                  [9, 7, 6, 8],
                  [8, 9, 5, 9],
                  [7, 8, 9, 10]]

Filas = len(calificaciones); #Aqui se esta contando la cantidad total de las filas
Columnas = len(calificaciones[0]); #Aqui se esta contando las columnas, desde la posicion 1

#Esto es para el nombre de las materias, queremos mostrar el array 
for i in range(Columnas+1):
	print(materias[i] , end="\t")
	
	if(i == Columnas):
		print()
	
		#Normal
for i in range(Filas): #filas
	print(nombres[i] , end = "\t" )
	sum=0
	
	for j in range(Columnas): #Columnas
		sum += calificaciones[i][j]
		
		print(" ",calificaciones[i][j], end="\t\t")
	print(sum/Columnas )
	 #Nueva linea después de cada fila
	
for j in range(Columnas): #columnas
	sum=0
	for i in range(Filas): #filas
		sum += calificaciones[i][j]
	print("  " ,sum/Filas, " ",end="\t" )




